src/Microsoft.Diagnostics.EventFlow.Core/Implementations/DiagnosticPipeline.cs

```diff
@@ -97,14 +110,24 @@ public class DiagnosticPipeline : IDisposable
 
             if (sinks.Count > 1)
             {
+                // After broadcasting we will effectively have (sinks.Count - 1) more events in the pipeline.
+                var eventCounter = new TransformBlock<EventData[], EventData[]>(
+                    (batch) => { Interlocked.Add(ref this.eventsInProgress, (sinks.Count - 1) * batch.Length);  return batch; },
```

Comment:
The broadcaster clones the events, that is why we need to increase the counter. Once you realize that it is actually pretty simple. I have updated the comment to make it clearer.